images/manageiq-base-worker/container-assets/manageiq_liveness_check

```diff
@@ -1,4 +1,17 @@
 #!/bin/bash
 
-[[ -s /etc/default/evm ]] && source /etc/default/evm
-ruby ${APP_ROOT}/lib/workers/bin/heartbeat_check.rb --heartbeat-file=${APP_ROOT}/tmp/worker.hb
+# We're alive if the hb_file exists and has a unix time "newer" / greater
+# than the current time.
+hb_file="${APP_ROOT}/tmp/worker.hb"
+if [[ -f ${hb_file} ]]; then
+  current_time=$(date +%s)
+  custom_timeout=$(cat ${hb_file})
+
+  if [ $current_time -le $custom_timeout ]; then
+    exit 0
+  else
+    exit 1
+  fi
```

Comment:
Similarly here

```suggestion
  if [ $current_time -gt $custom_timeout ]; then
    exit 1
  fi
```

The exit 0 will then naturally fall out.

Reply:
haha, I wanted to demonstrate the "normal" case exiting properly instead of a side effect of not exiting early.

Reply:
I can change it... I'm currently doing a build to make sure the two changes work together, so I can make any suggestions a bit later.

Reply:
haha, I just noticed the comment at the top will now match the code you're suggesting...

Nice work

```
# We're alive if the heartbeat_file exists and has a unix time "newer" / greater
# than the current time.
```

Reply:
Ok, changed.